File: Sep/Day5/minimum-operations-to-make-the-integer-zero.js
class Solution {
    makeTheIntegerZero(num1, num2) {
        for (let k = 1; ; k++) {
            let x = num1 - num2 * k;
            if (x < k) return -1;
            if (k >= x.toString(2).replace(/0/g, "").length) return k;
        }
    }
}

File: Sep/Day5/minimum-operations-to-make-the-integer-zero.java
class Solution {
    public int makeTheIntegerZero(int num1, int num2) {
        for (long k = 1; ; k++) {
            long x = (long) num1 - (long) num2 * k;
            if (x < k) return -1;
            if (k >= Long.bitCount(x)) return (int) k;
        }
    }
}
